src/Metadata/Resource/Factory/YamlResourceMetadataFactory.php

```
@@ -0,0 +1,147 @@
+ */
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Annotation\Resource;
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Symfony\Component\Yaml\Parser as YamlParser;
+
+/**
+ * Creates a resource metadata from yml {@see Resource} configuration.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class YamlResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $yamlParser;
+    private $decorated;
+    private $paths;
+
+    /**
+     * @param string[]                              $paths
+     * @param ResourceMetadataFactoryInterface|null $decorated
+     */
+    public function __construct(array $paths, ResourceMetadataFactoryInterface $decorated = null)
+    {
+        $this->yamlParser = new YamlParser();
+        $this->paths = $paths;
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass) : ResourceMetadata
+    {
+        $parentResourceMetadata = null;
+        if ($this->decorated) {
+            try {
+                $parentResourceMetadata = $this->decorated->create($resourceClass);
+            } catch (ResourceClassNotFoundException $resourceNotFoundException) {
+                // Ignore not found exception from decorated factories
+            }
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($resourceClass);
+        } catch (\ReflectionException $reflectionException) {
+            return $this->handleNotFound($parentResourceMetadata, $resourceClass);
+        }
+
+        $metadata = null;
+
+        foreach ($this->paths as $path) {
+            $resources = $this->yamlParser->parse(file_get_contents($path))['resources'];
+
+            foreach ($resources as $resource) {
+                if (!isset($resource['class']) || $resourceClass !== $resource['class']) {
```

It should throw an error instead of silently ignoring the problem.